check/performance

```diff
@@ -0,0 +1,15 @@
+#!/usr/bin/env bash
+
+################################################################################
+# Runs performance tests
+#
+# Usage:
+#     check/performance
+################################################################################
+
+# Get the working directory to the repo root.
+cd "$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+cd $(git rev-parse --show-toplevel)
+
+pytest --benchmark-autosave --benchmark-only --benchmark-compare --benchmark-compare-fail=min:10%  \
```

Comment:
IIUC, this makes the benchmark tests fail when minimum runtime of any test grows by more than 10%. Minimum statistic is very sensitive to occasional outliers. Let's make it more robust by using the mean instead.